assemblers/org.osbuild.qemu

```diff
@@ -172,17 +179,27 @@ def main(tree, output_dir, options, loop_client):
     # The partition table
     partitions = create_partition_table(image, options)
     partition_offset = partitions[0]["start"]
-    partition_size = partitions[0]["size"]
 
     # Create the level-2 bootloader
     install_grub2(image, grub2_fs_module, partition_offset)
 
-    with loop_client.device(image, partition_offset, partition_size) as loop:
-        # Populate the first partition of the image with a filesystem
-        mkfs(loop, root_fs_uuid)
-        # Copy the tree into the target image
-        with mount(loop) as mountpoint:
-            subprocess.run(["cp", "-a", f"{tree}/.", mountpoint], check=True)
+    # Now assemble the filesystem hierarchy and copy the tree into the image
+    with contextlib.ExitStack() as cm:
+        root = cm.enter_context(tempfile.TemporaryDirectory(prefix="osbuild-mnt"))
+        for partition in partitions:
+            offset, size = partition["start"], partition["size"]
+            filesystem = partition["filesystem"]
+            loop = cm.enter_context(loop_client.device(image, offset, size))
+            # make the specified filesystem
+            mkfs_for_type(loop, filesystem["uuid"], filesystem["type"])
+            # now mount it
+            mountpoint = filesystem["mountpoint"]
+            if mountpoint == "/":
+                mountpoint = root
+            cm.enter_context(mount(loop, root))
+        # the filesystem tree should now be properly setup,
+        # copy the tree into the target image
+        subprocess.run(["cp", "-a", f"{tree}/.", root], check=True)
```

Comment:
The metadata on the mountpoint will now be left unset (it is only applied to the root of the mounted fs). I suppose this is how things traditionally works, and that is the right way?